crates/nargo/src/ops/execute.rs

```diff
@@ -11,9 +13,39 @@ pub fn execute_circuit(
 ) -> Result<WitnessMap, NargoError> {
     let mut blocks = Blocks::default();
     let solver_status = solve(backend, &mut initial_witness, &mut blocks, circuit.opcodes)?;
-    if matches!(solver_status, PartialWitnessGeneratorStatus::RequiresOracleData { .. }) {
-        todo!("Add oracle support to nargo execute")
-        // TODO 1557
+
+    // TODO: Nargo only supports "oracle_print_impl" functions that print a singular value and nothing else
```

Comment:
link to issue

Reply:
This merges into Adam's branch so we can ignore and do cleanup there

Reply:
done